module/zfs/ddt_stats.c

```diff
@@ -138,20 +139,59 @@ ddt_get_dedup_object_stats(spa_t *spa, ddt_object_t *ddo_total)
 		for (ddt_type_t type = 0; type < DDT_TYPES; type++) {
 			for (ddt_class_t class = 0; class < DDT_CLASSES;
 			    class++) {
+				dmu_object_info_t doi;
+				uint64_t cnt;
+				int err;
+
+				/*
+				 * XXX this is a repeat of of the cached stats
+				 *     load in ddt_object_load(). feel like
+				 *     one of these isn't needed anymore, which
+				 *     is it? -- robn, 2024-02-07
+				 */
```

Comment:
Keeping both doesn't seem to terrible, but let's update this comment in `ddt_object_sync()` to indicate they're updated here as well.

```
        /*
         * Cache DDT statistics; this is the only time they'll change.
         */
```

Reply:
updated comment